client/my-sites/stats/features/modules/stats-top-posts/stats-top-posts.tsx

```diff
@@ -41,26 +44,26 @@ const StatsTopPosts: React.FC< StatsDefaultModuleProps > = ( {
 } ) => {
 	const translate = useTranslate();
 	const siteId = useSelector( getSelectedSiteId ) as number;
+	const optionLabels = useOptionLabels();
 
 	const isArchiveBreakdownEnabled: boolean = config.isEnabled( 'stats/archive-breakdown' );
 
 	const mainStatType = MAIN_STAT_TYPE;
 	const subStatType = SUB_STAT_TYPE;
 
-	const options: StatTypeOptionType[] = [
-		{
-			value: mainStatType,
-			label: translate( 'Post & pages' ),
-			mainItemLabel: translate( 'Posts & pages' ),
-		},
-		{
-			value: subStatType,
-			label: translate( 'Archive' ),
-			mainItemLabel: translate( 'Archive pages' ),
-		},
-	];
+	const options: StatTypeOptionType[] = Object.entries( optionLabels ).map( ( [ key, item ] ) => {
+		return {
+			value: key as StatType,
+			label: item.tabLabel,
+			mainItemLabel: item.mainItemLabel,
+		};
+	} );
 
 	const [ statType, setStatType ] = useState( mainStatType );
+	// Apply statType from the query to the component state to toggle the view list.
+	useEffect( () => {
+		setStatType( query.viewdType || mainStatType );
+	}, [ query.viewdType, mainStatType ] );
 	const onStatTypeChange = ( option: StatTypeOptionType ) => setStatType( option.value );
```

Comment:
```suggestion
	const [ localStatType, setLocalStatType ] = useState< StatType | null >( null );
	const statType = localStatType ?? query.viewdType ?? mainStatType;
	const onStatTypeChange = ( option: StatTypeOptionType ) => setLocalStatType( option?.value );
```

Is this the case of [You might not need an effect](https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state)?

Reply:
Good catch! Looks good to me. Updated in [34aa3f8](https://github.com/Automattic/wp-calypso/pull/104083/commits/34aa3f8832ce3f40973afb78002714450b4bec6c).